ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -131,21 +130,33 @@ private void setTxnConfigs() {
     }
   }
 
-  private boolean isValidTablesList(Tree tablesListTree) {
-    if (tablesListTree.getChildCount() <= 1) {
-      // If one or zero child, then it is valid.
-      // For single table replication, the valid format is <db_name>.<table_name>.
-      // To include multiple tables (t1 and t2), then valid format is <db_name>.[t1, t2].
-      // So, single child is always valid either it is table_name or tables_list.
-      return true;
-    }
-    assert(tablesListTree.getChildCount() == 2);
+  private void setReplDumpTablesList(Tree replTablesNode) throws HiveException {
+    assert(replTablesNode.getChildCount() <= 2);
+
+    // Traverse the children which can be either just include tables list or both include
+    // and exclude tables lists.
+    for (int listIdx = 0; listIdx < replTablesNode.getChildCount(); listIdx++) {
+      Tree tablesListNode = replTablesNode.getChild(listIdx);
+      assert(tablesListNode.getType() == TOK_REPL_TABLES_LIST);
+
+      List<String> tablesList = new ArrayList<>();
+      for (int child = 0; child < tablesListNode.getChildCount(); child++) {
```

Comment:
no need of index directly iterate through Tree type object

Reply:
We don't have any iterator or method that returns children Tree list. Need index to getChild.